code/game/machinery/gulag_teleporter.dm

```diff
@@ -0,0 +1,173 @@
+	B.apply_default_parts(src)
+	addtimer(src, "locate_reclaimer", 5)
+
+/obj/machinery/gulag_teleporter/Destroy()
+	if(linked_reclaimer)
+		linked_reclaimer.linked_teleporter = null
+	return ..()
+
+/obj/machinery/gulag_teleporter/power_change()
+	..()
+	update_icon()
+
+
+/obj/machinery/gulag_teleporter/interact(mob/user)
+	if(locked)
+		user << "[src] is locked."
+		return
+	toggle_open()
+
+/obj/machinery/gulag_teleporter/updateUsrDialog()
+	return
+
+/obj/machinery/gulag_teleporter/attackby(obj/item/I, mob/user)
+	if(!occupant && default_deconstruction_screwdriver(user, "[icon_state]", "[icon_state]",I))
+		update_icon()
+		return
+
+	if(default_deconstruction_crowbar(I))
+		return
+
+	if(default_pry_open(I))
+		return
+
+	return ..()
+
+/obj/machinery/gulag_teleporter/update_icon()
+	icon_state = initial(icon_state) + (state_open ? "_open" : "")
+	//no power or maintenance
+	if(stat & (NOPOWER|BROKEN))
+		icon_state += "_unpowered"
+		if((stat & MAINT) || panel_open)
+			icon_state += "_maintenance"
+		return
+
+	if((stat & MAINT) || panel_open)
+		icon_state += "_maintenance"
+		return
+
+	//running and someone in there
+	if(occupant)
+		icon_state += "_occupied"
+		return
+
+
+/obj/machinery/gulag_teleporter/relaymove(mob/user)
+	if(user.stat != CONSCIOUS)
+		return
+	if(locked)
+		user << "[src] is locked!"
+		return
+	open_machine()
+
+/obj/machinery/gulag_teleporter/container_resist(mob/living/user)
+	var/breakout_time = 600
+	if(!locked)
+		open_machine()
+		return
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user << "<span class='notice'>You lean on the back of [src] and start pushing the door open... (this will take about a minute.)</span>"
+	user.visible_message("<span class='italics'>You hear a metallic creaking from [src]!</span>")
+
+	if(do_after(user,(breakout_time), target = src))
+		if(!user || user.stat != CONSCIOUS || user.loc != src || state_open || !locked)
+			return
+
+		locked = FALSE
+		visible_message("<span class='warning'>[user] successfully broke out of [src]!</span>")
+		user << "<span class='notice'>You successfully break out of [src]!</span>"
+
+		open_machine()
+
+/obj/machinery/gulag_teleporter/proc/locate_reclaimer()
+	linked_reclaimer = locate(/obj/machinery/gulag_item_reclaimer)
+	if(linked_reclaimer)
+		linked_reclaimer.linked_teleporter = src
+
+/obj/machinery/gulag_teleporter/proc/toggle_open()
+	if(panel_open)
+		usr << "<span class='notice'>Close the maintenance panel first.</span>"
+		return
+
+	if(state_open)
+		close_machine()
+		return
+	if(!locked)
+		open_machine()
+
+// strips and stores all occupant's items
+/obj/machinery/gulag_teleporter/proc/strip_occupant()
+	if(linked_reclaimer)
+		linked_reclaimer.stored_items[occupant] = list()
+	for(var/obj/item/W in occupant)
+		if(occupant.unEquip(W))
+			if(istype(W, /obj/item/weapon/restraints/handcuffs))
+				W.forceMove(src)
```

Comment:
for the handcuffs, do you want to specifically put them on the floor instead so that sec officers can grab them back or is this intended behaviour were they get stored in the teleporters contents?